rq/dependency.py

```diff
@@ -0,0 +1,25 @@
+from .job import Job
+from redis.client import Pipeline
+from typing import List
+from redis.exceptions import WatchError
+
+
+class Dependency:
+    @classmethod
+    def get_jobs_with_met_dependencies(cls, jobs: List['Job'], pipeline: Pipeline):
+        jobs_with_met_dependencies = []
+        jobs_with_unmet_dependencies = []
+        for job in jobs:
+            while True:
+                try:
+                    pipeline.watch(*[Job.key_for(dependency_id) for dependency_id in job._dependency_ids])
+                    job.register_dependency(pipeline=pipeline)
+                    if job.dependencies_are_met(pipeline=pipeline):
+                        jobs_with_met_dependencies.append(job)
```

Comment:
Hey sorry but it seems like this logic branch is not tested. To hit this branch, we need to call `enqueue_many()` with jobs whose dependencies are met.

Sorry for being fussy, hopefully this will be the last review.

Reply:
Just pushed a change to the test. Really appreciate your help with this!